src/librbd/crypto/DataCryptor.h

```diff
@@ -0,0 +1,38 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#ifndef CEPH_LIBRBD_CRYPTO_DATA_CRYPTOR_H
+#define CEPH_LIBRBD_CRYPTO_DATA_CRYPTOR_H
+
+#include "include/int_types.h"
+
+namespace librbd {
+namespace crypto {
+
+enum cipher_mode_t {
```

Comment:
Nit: I'd recommend moving this to a `librbd/crypto/Types.h` file and renaming to `CipherMode` with the enums being explicitly named like `CIPHER_MODE_ENC/DEC`